packages/gcc7.rb

```diff
@@ -74,20 +84,49 @@ def self.build
                  "--enable-languages=c,c++,fortran",
                  "--disable-multilib",
                  "--enable-threads=posix",
-                 "--disable-bootstrap",
+                 "--enable-bootstrap",
+                 "--disable-werror",
                  "--disable-libmpx",
                  "--disable-static",
                  "--program-suffix=-7.3",
-                 "--with-arch-32=i686"
+                 "--with-arch-32=i686",
+                 "--with-default-libstdcxx-abi=gcc4-compatible"
       end
-        system "make"
+      # Comment: --with-default-libstdcxx-abi=gcc-compatible
+      #          Use this switch if we are upgrading from GCC version prior to 5.1.0
+      #          We do not want to recompile all the libraries written in C++
+      
+      system "make"
     end
   end
-
-  def self.install
+  
+  
+  # preserve for check, skip check for current version
+  def self.check
     Dir.chdir("objdir") do
-	    system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
+      #system "ulimit -s 32768"
+      #system "make -k check -j8"
+      #system "../contrib/test_summary > ~/Downloads/gcc_test_summary.txt"
     end
   end
-    
+  
+  def self.install
+    Dir.chdir("objdir") do
+      system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
+	   
+      # http://www.linuxfromscratch.org/lfs/view/development/chapter06/gcc.html#contents-gcc
+      # move a misplaced file
+      # The installation stage puts some files used by gdb under the /usr/local/lib(64) directory. This generates spurious error messages when performing ldconfig. This command moves the files to another location.
+      system "mkdir -pv #{CREW_DEST_PREFIX}/share/gdb/auto-load/usr/lib"
+      system "mv -v #{CREW_DEST_LIB_PREFIX}/*gdb.py #{CREW_DEST_PREFIX}/share/gdb/auto-load/usr/lib"
+	    
+      # Install Binary File Descriptor library (BFD)
+      system "install -v -dm755 #{CREW_DEST_LIB_PREFIX}/bfd-plugins"
+      
+      # Add a compatibility symlink to enable building programs with Link Time Optimization (LTO)
+      system "ln -sfv #{CREW_DEST_PREFIX}/libexec/gcc/$(gcc -dumpmachine)/7.3.0/liblto_plugin.so #{CREW_DEST_LIB_PREFIX}/bfd-plugins/"
+      system "ln -sv gcc-7.3 #{CREW_DEST_PREFIX}/bin/gcc"
+      system "ln -sv gcc-7.3 #{CREW_DEST_PREFIX}/bin/cc"
```

Comment:
Did you create new pre-built binaries after these changes?

Reply:
newly recompiled